packages/desktop-client/src/components/modals/manager/DuplicateFileModal.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+
+import { addNotification, duplicateBudget } from 'loot-core/client/actions';
+import { type File } from 'loot-core/src/types/file';
+
+import { theme } from '../../../style';
+import { Button, ButtonWithLoading } from '../../common/Button2';
+import { FormError } from '../../common/FormError';
+import { InitialFocus } from '../../common/InitialFocus';
+import { InlineField } from '../../common/InlineField';
+import { Input } from '../../common/Input';
+import {
+  Modal,
+  ModalButtons,
+  ModalCloseButton,
+  ModalHeader,
+} from '../../common/Modal';
+import { Text } from '../../common/Text';
+import { View } from '../../common/View';
+
+type DuplicateFileProps = {
+  file: File;
+  managePage?: boolean;
+  loadBudget?: 'none' | 'original' | 'copy';
+  onComplete?: (event: {
+    status: 'success' | 'failed' | 'canceled';
+    error?: object;
+  }) => void;
+};
+
+export function DuplicateFileModal({
+  file,
+  managePage,
+  loadBudget = 'none',
+  onComplete,
+}: DuplicateFileProps) {
+  const { t } = useTranslation();
+  const [newName, setNewName] = useState(file.name + ' - copy');
+  const [nameError, setNameError] = useState<string | null>(null);
+
+  // If the state is "broken" that means it was created by another user.
+  const isCloudFile = 'cloudFileId' in file && file.state !== 'broken';
+  const dispatch = useDispatch();
+
+  const [loadingState, setLoadingState] = useState<'cloud' | 'local' | null>(
+    null,
+  );
+
+  const validateNewName = (name: string): string | null => {
+    const trimmedName = name.trim();
+    if (trimmedName === '') return t('Budget name cannot be blank');
+    if (trimmedName.length > 100) {
+      return t('Budget name is too long (max length 100)');
+    }
+    if (!/^[a-zA-Z0-9 .\-_()]+$/.test(trimmedName)) {
```

Comment:
Did you take this from somewhere? We don't have this logic in the ```BudgetName``` file so i'm tempted to remove it for consistency. Ideally we wouldn't validate in one place and not another.

Reply:
I'm ok with removing it, but want to list my findings and concerns.

Currently (release and edge) don't have a problem with validating a budget name because the filename (really the directory name) will {almost} always be one of three options ```Test-Budget-xxxxxx```, ```My-Finances-xxxxxx```, or ```Demo-Budget```.  The exception to this is when a budget is imported, then it will have a different directory name.  Once the directory name is created, it stays the same no matter what the budget is renamed to.  If I create a new budget it will get the directory name of ```My-Finances-xxxxxx``` and inside that directory will be the db.sqlite file that holds the budget plus a metadata.json file that holds the name of the budget.  So, if I rename that budget, the directory stays ```My-Finances-xxxxxx``` but the budget name inside the metadata changes.

Now, moving to Duplicating.  When you enter a budget name into the Duplicate modal, that budget name will become the directory name.  Each filesystem has it's own validation requirements for file/directory name.  I tried to use as generic of validations as possible to work on as many filesystems as possible.  The actual characters aren't that big of a deal as the ```idFromFileName``` will change any non-alphanumeric character to a '-', but the length could be an issue.  I added the length validation and then coderabbitai recommended the character validation.

The web-app won't have an issue with the directory name length because it is using IndexedDB as a filesystem and doesn't have the length or character limits.  However, since this PR will apply to both the web-app and electron, I figured validation was needed.

Yes, I think it would be better to be shared outside DuplicateFileModal.  I've been trying to figure out a way to do both budget name validation and unique name checking (```uniqueFileName``` from packages/loot-core/src/server/util/budget-name.ts) in both the client and server, but there's some differences between the client and server side in what functions can be used to get the list of filename.

Those are my thoughts.  I can continue to figure out a way to share the validation or I can remove it altogether.

Reply:
OK, I get you. I think the character regex wont work for Chinese characters etc, I'd be tempted to remove the character validation until we can figure out what we should support across the board.

![image](https://github.com/user-attachments/assets/77f35a27-954b-4f2a-b522-226a2c3ea89c)

Reply:
Moved budget name validation from DuplicateFileModal to loot-core/server, with client side actions.
Removed character validation.
Added ability to get uniqueBudgetName on the client side, so now the New Budget Name input box is prefilled with a unique name.
Did not add validation to rename budget. I think that can be a separate PR, unless you'd like that added in now.